src/Cli/AzureCliLogin.ts

```diff
@@ -24,39 +30,35 @@ export class AzureCliLogin {
                 }
             }
         };
-        await this.executeAzCliCommand("--version", true, execOptions);
-        core.debug(`az cli version used:\n${output}`);
+
+        await this.executeAzCliCommand(["--version"], true, execOptions);
+        core.debug(`Azure CLI version used:\n${output}`);
 
         this.setAzurestackEnvIfNecessary();
 
-        await this.executeAzCliCommand(`cloud set -n "${this.loginConfig.environment}"`, false);
-        console.log(`Done setting cloud: "${this.loginConfig.environment}"`);
+        await this.executeAzCliCommand(["cloud", "set", "-n", this.loginConfig.environment], false);
+        core.info(`Done setting cloud: "${this.loginConfig.environment}"`);
 
-        // Attempting Az cli login
-        var commonArgs = ["--service-principal",
-            "-u", this.loginConfig.servicePrincipalId,
-            "--tenant", this.loginConfig.tenantId
-        ];
-        if (this.loginConfig.allowNoSubscriptionsLogin) {
-            commonArgs = commonArgs.concat("--allow-no-subscriptions");
-        }
-        if (this.loginConfig.enableOIDC) {
-            commonArgs = commonArgs.concat("--federated-token", this.loginConfig.federatedToken);
+        if (this.loginConfig.authType == "service_principal") {
+            let args = ["--service-principal",
+                "--username", this.loginConfig.servicePrincipalId,
+                "--tenant", this.loginConfig.tenantId
+            ];
+            if (this.loginConfig.servicePrincipalKey) {
+                await this.loginWithSecret(args);
+            }
+            else {
+                await this.loginWithOIDC(args);
+            }
         }
         else {
-            console.log("Note: Azure/login action also supports OIDC login mechanism. Refer https://github.com/azure/login#configure-a-service-principal-with-a-federated-credential-to-use-oidc-based-authentication for more details.")
-            commonArgs = commonArgs.concat(`--password=${this.loginConfig.servicePrincipalKey}`);
-        }
-
-        const loginOptions: ExecOptions = defaultExecOptions();
-        await this.executeAzCliCommand(`login`, true, loginOptions, commonArgs);
-
-        if (!this.loginConfig.allowNoSubscriptionsLogin) {
-            var args = [
-                "--subscription",
-                this.loginConfig.subscriptionId
-            ];
-            await this.executeAzCliCommand(`account set`, true, loginOptions, args);
+            let args = ["--identity"];
+            if (this.loginConfig.servicePrincipalId) {
+                await this.loginWithUserAssignedIdentity(args);
+            }
+            else {
+                await this.loginWithSystemAssignedIdentity(args);

```

Comment:
Can `loginWithSystemAssignedIdentity` function be written inline as it only contains 1 line and is called only once?